portfolio/src/main/webapp/scripts/project.js

```diff
@@ -0,0 +1,54 @@
+/** Represents a project in 'Featured Works' of the main page. */
+class Project {
+    constructor(name, image, description, tools, date, page) {
+      ensureNonNull(name, description, tools, date, page);
+  
+      const div = document.createElement('div');
+      const divClasses = ['project', 'hide-project'];
+      divClasses.forEach(divClass => div.classList.add(divClass));
+  
+      const divContent = `<div class="project-name-placeholder">
+                            <div class="project-image-placeholder">
+                              <h1 class="project-name">${name}</h1>
+                            </div>
+                          </div>
+                          <div class="project-details-placeholder">
+                            <p class="project-description">${description}</p>
+                            <p class="project-tools">Made with: ${tools}</p>
+                            <p class="project-date">Done: ${date}</p>
+                            <div class="button more-button"><a href="${page}">More >></a></div>
+                          </div>`;   
+      div.innerHTML = divContent;
+      
+      this.addProjectImage(div, image);
+      this.projectCondensed = div;
+    }
+  
+    addProjectImage(element, image) {
+      if (image === '') {
```

Comment:
I guess this image is actually `imagePath`, as we expect it to be an empty string? If that's the case, please update the naming for `image` in these files :)